app/components/market/submit/side-menu.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, Input, OnChanges, OnDestroy } from "@angular/core";
+import { FormGroup } from "@angular/forms";
+import { NcjPoolTemplate, ServerError } from "app/models";
+import { PricingService, PythonRpcService } from "app/services";
+import { NumberUtils } from "app/utils";
+import { Subscription } from "rxjs/Subscription";
+import { Modes, NcjParameterWrapper } from "./market-application.model";
+import "./side-menu.scss";
+
+@Component({
+    selector: "bl-side-menu",
+    templateUrl: "side-menu.html",
+})
+export class SideMenuComponent implements OnChanges, OnDestroy {
+    public static breadcrumb() {
+        return { name: "Sidemenu" };
+    }
+    public Modes = Modes;
+    public estimatedCost = "n/a";
+    public modeText: string[] = ["None", "Run job with auto pool",
+        "Run job with existing pool", "Create pool for later use"];
+    public error: ServerError;
+    @Input() public logo: string;
+    @Input() public title: string;
+    @Input() public action: string;
+    @Input() public modeState: number;
+    @Input() public form: FormGroup;
+    @Input() public poolParametersWrapper: NcjParameterWrapper[];
+    @Input() public jobParametersWrapper: NcjParameterWrapper[];
+    @Input() public poolTemplate: NcjPoolTemplate;
+    private _subs: Subscription[] = [];
+
+    constructor(private pricingService: PricingService,
+                private pythonRpcService: PythonRpcService) {
+        // do nothing
+    }
+
+    public ngOnChanges(changes) {
+        if (changes.form) {
+            this._subs.push(this.form.valueChanges.subscribe((value) => {
```

Comment:
you need to unsubscribe here before you do that. Also for that case don't use a list of subscription(1st it is only used by one element, 2nd it need to be cleared everytime the form changes which might not be the case of other subscirption you might want to add later